### DISCUSSION TITLE
2.0 同一份 schema 渲染多个 field 的场景(array.items) 改其中一个 field.componentProps.xxx = xxx 导致所有 field 一起变化

### DISCUSSION THREAD

**Discussion by linfaxin** [alibaba/formily#1426] [Q&A 2.x] [answered]
期望：从同一份 schema 里渲染出来的 field 应该是独立的，各自的 field.componentProps 应该也是相互隔离的。
现状：不隔离。
看如何修复合适

![image](https://user-images.githubusercontent.com/3705017/118217394-f7829380-b4a7-11eb-810b-4bff8607fdd3.png)
![image](https://user-images.githubusercontent.com/3705017/118217450-13863500-b4a8-11eb-8081-f3f6f4caeafd.png)
![image](https://user-images.githubusercontent.com/3705017/118217468-1da83380-b4a8-11eb-8b91-e5db1b14f202.png)

**Comment by janryWang** [comment]
初步想法是给ArrayTable这些组件在递归渲染的时候单独深拷贝schema，其实在1.x中就是这样做的，具体实现还得想想

**Comment by janryWang** [comment, marked as answer]
已经修复了，主要解决思路是本身compiler就是一次深度递归编译，所以就在toFieldProps的时候，直接复用compiled的Schema即可
